auctions/dutch-auction/src/abi.sw

```diff
@@ -0,0 +1,16 @@
+library abi;
+
+dep data_structures;
+
+use std::{contract_id::ContractId, identity::Identity};
+
+use data_structures::Auction;
+
+abi DutchAuction {
+    fn price(auction_id: u64) -> u64;
+    fn bid(auction_id: u64);
+    fn create_auction(opening_price: u64, reserve_price: u64, start_time: u64, end_time: u64, beneficiary: Identity, asset: ContractId);
```

Comment:
Despite the conversation somewhere else in the PR we still need a function that accepts an Identity as a parameter and returns their currently active auctions and completed auctions. This would be a struct with each field being an array for now -> vec in the future

Reply:
How would i do this with an array? Only vec seems plausible

Reply:
My workaround is an array with 1 item.

```rust
struct SomeName {
    active: [u64; 1],
    completed: [u64; 1]
}
```

Then you replace the entire array when needed. It's not ideal for correct functionality so you could make it larger and add some boilerplate to handle indexing etc. but that's not really needed. As long as 1 works in the UI then it's fine imo until vec is out

Reply:
okay just added it with a array for now then, added some warnings saying its temporary till a vec is out